src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -73,18 +74,18 @@ class LogInWithShortLivedTokenPage extends Component {
 
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+
+        // We have two routes in our history, "/" and "transition". We want to dismiss the trasition route, remove it from the history
+        // or else a user could be routed back to `/transition` when they close the modal.
+        // Or in some cases, the home screen gets stuck in `/transition` and there is an endless spinner in the background
```

Comment:
What cases like this remain? The endless spinner we saw is from this page. And we figured out that this occurred because someone removed the code that should dismiss it.

Reply:
Right, so I was just explaining why we needed to move dismissModal to before this if condition. If dismiss modal was placed after the early return for workspace/new, it would result in an endless spinner